### PR TITLE
profiles: restructure amd64, arm and x86 musl profiles

### PR DESCRIPTION
**Pull request by blueness**
This restructuring brings the musl profiles into conformity with the glibc profiles for amd64, arm and x86.  Yet to be done are the profiles for ppc and mips.

Signed-off-by: Anthony G. Basile <blueness@gentoo.org>

### CODE REVIEW

File: profiles/default/linux/amd64/17.0/musl/make.defaults [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ARCH="amd64"
+ACCEPT_KEYWORDS="${ARCH}"
+
+CHOST="x86_64-gentoo-linux-musl"
+CFLAGS="-O2 -pipe"
+CXXFLAGS="${CFLAGS}"
+FFLAGS="${CFLAGS}"
+FCFLAGS="${CFLAGS}"
+
+# Anthony G. Basile <blueness@gentoo.org> (01 Jul 2014)
```
Comment (mgorny) [inline profiles/default/linux/amd64/17.0/musl/make.defaults]: Isn't most of this redundant when you use arch/amd64?

File: profiles/features/musl/package.use.mask [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+# llvm's sanitizers are also incompatible with musl
+sys-devel/llvm sanitize
+
+# These cause collisions with <libintl.h>
+# even with --without-included-gettext
+sys-devel/gettext nls
+sys-fs/e2fsprogs nls
+
+# Broken
+dev-vcs/git gpg
```
Comment (stefson) [inline profiles/features/musl/package.use.mask]: Wasn't that fixed with the stabilization of >gnupg-2.2?

File: profiles/default/linux/amd64/17.0/musl/make.defaults (hunk 1 above)
Comment (blueness) [reply profiles/default/linux/amd64/17.0/musl/make.defaults]: it may be, i'll check.

File: profiles/features/musl/package.use.mask (hunk 2 above)
Comment (blueness) [reply profiles/features/musl/package.use.mask]: yes it was, thanks for catching that.

**Comment by blueness** [comment]
@mgorny okay i checked and cleaned out the redundancy with arch/amd64 wrt make.defaults.  If there are no more problems, I'll commit.

**Comment by stefson** [comment]
It would be great if you could issue a news file in case this requires any work from the user.

File: profiles/default/linux/amd64/17.0/musl/make.defaults [hunk 3]
```diff
@@ -0,0 +1,10 @@
+# Copyright 1999-2015 Gentoo Foundation.
```
Comment (mgorny) [inline profiles/default/linux/amd64/17.0/musl/make.defaults]: Sorry for not noticing this earlier but I think you need to update your calendar ;-).

File: profiles/default/linux/amd64/17.0/musl/make.defaults [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+# Distributed under the terms of the GNU General Public License v2
+
+CHOST="x86_64-gentoo-linux-musl"
+
+# Anthony G. Basile <blueness@gentoo.org> (01 Jul 2014)
+# Multilib-related setup, bug #515130
+MULTILIB_ABIS="amd64"
```
Comment (mgorny) [inline profiles/default/linux/amd64/17.0/musl/make.defaults]: Doesn't no-multilib profile set that? Or does some other profile override it?

File: profiles/default/linux/amd64/17.0/musl/package.use.force [hunk 5]
```diff
@@ -0,0 +1,7 @@
+# Copyright 1999-2016 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+# Michał Górny <mgorny@gentoo.org> (24 Sep 2016)
```
Comment (mgorny) [inline profiles/default/linux/amd64/17.0/musl/package.use.force]: Doesn't amd64 profile include that?

File: profiles/default/linux/amd64/17.0/musl/use.force [hunk 6]
```diff
@@ -0,0 +1,6 @@
+# Copyright 1999-2014 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+# Force the flag which corresponds to arch/abi, bug #515130
+amd64
```
Comment (mgorny) [inline profiles/default/linux/amd64/17.0/musl/use.force]: Likewise.

File: profiles/default/linux/amd64/17.0/musl/use.mask [hunk 7]
```diff
@@ -0,0 +1,6 @@
+# Copyright 1999-2017 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+# Force the flag which corresponds to arch/abi, bug #515130
```
Comment (mgorny) [inline profiles/default/linux/amd64/17.0/musl/use.mask]: Likewise.

File: profiles/default/linux/arm/17.0/musl/parent [hunk 8]
```diff
@@ -0,0 +1,3 @@
+..
+../../../../../arch/amd64/no-multilib
```
Comment (mgorny) [inline profiles/default/linux/arm/17.0/musl/parent]: Do you really want arm to inherit from amd64?

File: profiles/default/linux/arm/17.0/musl/make.defaults [hunk 9]
```diff
@@ -0,0 +1,4 @@
+# Copyright 1999-2014 Gentoo Foundation.
+# Distributed under the terms of the GNU General Public License v2
+
+CHOST="arm-unknown-linux-musleabi"
```
Comment (mgorny) [inline profiles/default/linux/arm/17.0/musl/make.defaults]: You want to explicitly set CHOST_arm too.

File: profiles/default/linux/amd64/17.0/musl/make.defaults (hunk 4 above)
Comment (blueness) [reply profiles/default/linux/amd64/17.0/musl/make.defaults]: I had to force it.

File: profiles/default/linux/amd64/17.0/musl/package.use.force (hunk 5 above)
Comment (blueness) [reply profiles/default/linux/amd64/17.0/musl/package.use.force]: yes it does.  its gone.

File: profiles/default/linux/amd64/17.0/musl/use.force (hunk 6 above)
Comment (blueness) [reply profiles/default/linux/amd64/17.0/musl/use.force]: yep.  i got so used to running the whole show myself, i'm not even familiar what's in the other stack layers.

File: profiles/default/linux/amd64/17.0/musl/use.mask (hunk 7 above)
Comment (blueness) [reply profiles/default/linux/amd64/17.0/musl/use.mask]: ditto

File: profiles/default/linux/arm/17.0/musl/make.defaults (hunk 9 above)
Comment (blueness) [reply profiles/default/linux/arm/17.0/musl/make.defaults]: done

File: profiles/default/linux/arm/17.0/musl/parent (hunk 8 above)
Comment (blueness) [reply profiles/default/linux/arm/17.0/musl/parent]: nope, that's a typo

**Comment by blueness** [comment]
@mgorny some of the other redundancy you not above applies to arm and x86 as well.  i'll clean those up

**Comment by blueness** [comment]
pushed

**Pull request closed without merging**